### ISSUE TITLE
fails to decompress some MMCMP files

### ISSUE THREAD

**Issue by manxorist** [temisu/ancient#20]
<https://manx.datengang.de/openmpt/temp/mmcmp.zip> contains 3 files, only mmcmp-ELECTR~1.MOD is decompressed successfully by ancient.

They have been compressed with MMCMP 1.34 (available here: <http://cd.textfiles.com/scene96-2/programs/mmcmp134/>).

```
manx@quadratus:~/projects/ancient/ancient.git$ ./ancient decompress /home/manx/c-home/stuff/mmcmp/compressed/BENJAM.IT /home/manx/c-home/stuff/mmcmp/ancient/BENJAM.IT
Decompression failed for /home/manx/c-home/stuff/mmcmp/compressed/BENJAM.IT
manx@quadratus:~/projects/ancient/ancient.git$ ./ancient decompress /home/manx/c-home/stuff/mmcmp/compressed/MELTED.XM /home/manx/c-home/stuff/mmcmp/ancient/MELTED.XM
Verify (raw) failed for /home/manx/c-home/stuff/mmcmp/compressed/MELTED.XM
manx@quadratus:~/projects/ancient/ancient.git$ ./ancient decompress /home/manx/c-home/stuff/mmcmp/compressed/mmcmp-ELECTR~1.MOD /home/manx/c-home/stuff/mmcmp/ancient/mmcmp-ELECTR~1.MOD
manx@quadratus:~/projects/ancient/ancient.git$
```

OpenMPT/libopenmpt can decompress them successfully and match mmuncmp output exactly. 

Our current implementation (BSD-3-Clause licensed, so only as reference): <https://github.com/OpenMPT/openmpt/blob/master/soundlib/ContainerMMCMP.cpp>.

I am no expert on actual compression formats, so I have not looked in any further detail.

**Comment by temisu** [comment]
These were amazing test files, thank you for that. I'll add them to my test suite.

First problem was due to flag processing which I tried to "fix and improve" compared from MMCMP. I was obviously trying to be too clever (MELTED.XM)

The second file (BENJAM.IT) was really interesting: It has buffer overflow in it. The library rightfully threw an exception on attempted illegal access. However, I modified the code that instead of failing it will cut the output short. This fixed the issue. (Obviously any decompressor that does not handle this case gracefully has security issues...)
